src/lib/y2partitioner/format_mount/base.rb

```diff
@@ -43,16 +46,15 @@ def apply_options!
       def apply_format_options!
         return false unless @options.encrypt || @options.format
 
-        @partition.remove_descendants
+        partition.remove_descendants
 
-        if @options.encrypt
-          @partition = @partition.create_encryption("cr_#{@partition.basename}")
-          @partition.password = @options.password
+        if encrypt?
+          @partition = partition.create_encryption("cr_#{@partition.basename}")
```

Comment:
We should probably create a PBI for correct and robust generation of names for the crypt devices, both in the partitioner and in the proposal.